src/components/icons/LottieAnimation.tsx

```diff
@@ -1,36 +1,128 @@
 import Player, { LottieRefCurrentProps } from 'lottie-react'
-import React, { useEffect, useRef } from 'react'
+import React, { useCallback, useEffect, useRef, useState } from 'react'
 
 interface LottieAnimationProps {
   animationData: any
   speed?: number
+  color?: string
   onComplete?: () => void
   style?: React.CSSProperties
 }
 
 /**
  * LottieAnimation Component.
  *
- * This component renders a Lottie animation using the `lottie-react` Player.
- * It accepts animation data and an optional speed prop to control the animation speed.
+ * This component utilizes the 'lottie-react' Player to render a Lottie animation.
+ * It accepts animation data and supports customizable speed and color properties,
+ * allowing for dynamic control over the animation's playback rate and the stroke
+ * and fill colors of its elements.
  *
+ * Props:
+ * animationData: The JSON data that defines the Lottie animation.
+ * Speed: Optional. A number that specifies the playback speed of the animation. Defaults to 1 for normal speed.
+ * Color: Optional. A string representing the hex color code used to update the stroke and fill colors of the animation. Defaults to '#FFFFFF' (white) if not specified.
+ * OnComplete: Optional. A callback function that is triggered when the animation completes its playback.
+ * Style: Optional. A React.CSSProperties object to apply inline styles to the player container.
+ * This does not affect the Lottie animation's JSON content or alter its visual elements.
  */
 const LottieAnimation: React.FC<LottieAnimationProps> = ({
   animationData,
-  speed = 1, // Default speed set to 1x
+  speed = 1,
+  color = '#FFFFFF',
   onComplete,
   style,
 }) => {
+  const [key, setKey] = useState(0)
   const lottieRef = useRef<LottieRefCurrentProps | null>(null)
 
+  /**
+   * Converts hex color to RGBA.
+   *
+   * @param hex - The hex color code.
+   * @returns The RGBA values.
+   */
+  const hexToRGBA = useCallback((hex: string): number[] => {
+    let r = 0,
+      g = 0,
+      b = 0
+    const a = 1
+    if (hex.length === 7) {
+      r = parseInt(hex.slice(1, 3), 16) / 255
+      g = parseInt(hex.slice(3, 5), 16) / 255
+      b = parseInt(hex.slice(5, 7), 16) / 255
+    }
+    return [r, g, b, a]
+  }, [])
+
+  /**
+   * Updates colors in a single shape item.
+   *
+   * This function checks if the item represents a stroke or
+   * fill and updates its color using the provided RGBA array.
+   * If the item contains a nested items array, it recursively
+   * calls itself to ensure that all relevant shapes within the
+   * structure are updated with the new color.
+   *
+   * @param item - The shape item object which may contain
+   * stroke or fill color properties.
+   * @param rgbaArray - The RGBA values to set as new color.
+   */
+  const updateColorsInItem = useCallback((item: any, rgbaArray: number[]) => {
```

Comment:
This function could also be declared outside of the scope of the component without `useCallback`.

Reply:
@trevinhofmann, can we move these functions to utils? What do you think?

Reply:
For now, I would be okay with either keeping them in the component file (outside of the component definition) or moving them to the `util` directory. If it is likely that they would be reusable in other files, then it would definitely make sense to move them to the `util` directory.

Reply:
I think you can use `hexToRgb` provided by `@mui/material` which is already included in the project.

---

FYI I follow a simple rule when it comes to helper functions: If it's only used in one file, define it in that file. If it's used by multiple files, define it in the `util` directory.

This reduces surface area and provides a more deterministic rule-of-thumb that avoids developers having to anticipate future needs.